File: app/src/main/java/ch/abler/aline/alarmdroid/syncadapter/AuthenticatorActivity.java
package ch.abler.aline.alarmdroid.syncadapter;

import android.accounts.Account;
import android.accounts.AccountAuthenticatorActivity;
import android.accounts.AccountManager;
import android.content.ContentResolver;
import android.content.Intent;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import org.w3c.dom.Text;

import ch.abler.aline.alarmdroid.R;

public class AuthenticatorActivity extends AccountAuthenticatorActivity implements View.OnClickListener{

    public static String EXTRA_ACC_TYPE = "acctype";
    public static String EXTRA_AUTH_TYPE = "authtype";
    public static String EXTRA_IS_NEW_ACC = "isnewacc";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_authenticator);

        Button btn = (Button) findViewById(R.id.accountSubmitButton);
        btn.setOnClickListener(this);
        debug(btn.getText().toString());

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        toolbar.setTitle(R.string.app_name);
    }

    private void submit() {
        debug("submitting acc data");
        TextView serverAddress = (TextView) findViewById(R.id.serverAddress);
        TextView serverPort = (TextView) findViewById(R.id.serverPort);
        TextView accountName = (TextView) findViewById(R.id.accountName);
        final String address = serverAddress.getText().toString();
        final String port = serverPort.getText().toString();
        final String name = accountName.getText().toString();

        final Account account = new Account(name, Constants.ACCOUNT_TYPE);

        AccountManager am = AccountManager.get(this);

        Bundle userdata = new Bundle();

        userdata.putString(Constants.KEY_SERVER_ADDR, address);
        userdata.putString(Constants.KEY_SERVER_PORT, port);

        am.addAccountExplicitly(account,null,userdata);


        // set up account syncing
        ContentResolver.addPeriodicSync(account, Constants.AUTHORITY, Bundle.EMPTY, Constants.SYNC_INTERVAL);

        setAccountAuthenticatorResult(null);
        setResult(RESULT_OK);
        finish();
    }

    @Override
    public void onClick(View view) {
        submit();
    }

    private void debug(String msg){
        Log.d("AuthenticatorActivity", msg);
    }
}

File: app/src/main/java/ch/abler/aline/alarmdroid/AuthenticatorActivity.kt
package ch.abler.aline.alarmdroid

import android.accounts.Account
import android.accounts.AccountAuthenticatorActivity
import android.accounts.AccountManager
import android.content.ContentResolver
import android.content.SyncRequest
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText

class AuthenticatorActivity : AccountAuthenticatorActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_authenticator)
        accountManager = AccountManager.get(this)

        saveButton = findViewById(R.id.submit_account_button)
        nameInput = findViewById(R.id.account_name_input)
        urlInput = findViewById(R.id.url_input)

        saveButton.setOnClickListener(this)
    }

    private fun submit() {
        val name = nameInput.text.toString()
        val url = urlInput.text.toString()
        val account = Account(name, ACCOUNT_TYPE)

        val userdata = Bundle()

        userdata.putString(KEY_SYNC_URL, url)

        accountManager.addAccountExplicitly(account, null, userdata)

        ContentResolver.setIsSyncable(account, AUTHORITY, 1)
        ContentResolver.setSyncAutomatically(account, AUTHORITY, true)

        ContentResolver.requestSync(SyncRequest.Builder()
            .syncPeriodic(SYNC_INTERVAL, 600)
            .setSyncAdapter(account, AUTHORITY)
            .setExtras(Bundle.EMPTY)
            .build()
        )

        Log.d("AuthenticatorActivity", "Account added")

        setAccountAuthenticatorResult(Bundle.EMPTY)
        setResult(RESULT_OK)
        finish()
    }

    override fun onClick(v: View?) {
        submit()
    }

    companion object {
        lateinit var saveButton: Button
        lateinit var nameInput: EditText
        lateinit var urlInput: EditText

        lateinit var accountManager: AccountManager

        const val ACCOUNT_TYPE = "ch.abler.aline.alarmdroid"
        const val KEY_SYNC_URL  = "ch.abler.aline.alarmdroid.sync_url"
        const val AUTHORITY = "ch.abler.aline.alarmdroid.provider"
        const val SYNC_INTERVAL = 60L * 60 // sync every 60 minutes
    }
}